src/v2i-hub/docs/Programming_Guide.md

```diff
@@ -0,0 +1,1233 @@
+# Programming Guide
+
+The V2X Hub Programming Guide was developed during the V2I Reference Implementation project for developers of plugins for the V2X Hub in supporting Connected Vehicle (CV) technology. This programming guide gives details on how to create a plugin for the V2X Hub based on the Example Plugin created by Battelle. The Example Plugin can be obtained from OSADP, along with the other V2X Hub software that you will need to compile and setup to do your own V2X Hub development.
+
+The V2X Hub consists of software modules written in C and C++ for Linux. The developer wanting to create a plugin for the V2X Hub must be familiar with C, C++ and Linux, and have a basic knowledge of object-oriented design. Some other important software development practices to be familiar with are threading, network communications, external library integration, and data serialization just to name a few. Development is done using the following tools: cmake, make, g++ compiler, and optionally eclipse. Developers should have familiarity with those tools to successfully create a V2X Hub plugin.
```

Comment:
Do we use eclipse?

Reply:
This is a great catch. I should link this to our VSCode DevContainer setup